### ISSUE TITLE
[AI] Recruit Rushers CA throws Lua Error and then Recruitment for Side terminates

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#8297]
### Game and System Information

- **Version:** 1.17.25+dev
- **Downloaded from:** Github
- **Build info:** msys2 compiler
- **OS:** Windows 10

### Description of the bug

On the side with Experimental AI enabled, this popped up and then side did not even recruit a single unit.
```txt
20240126 01:12:47 error scripting/lua: ai/lua/ca_recruit_rushers.lua:155: attempt to perform arithmetic on a nil value
stack traceback:
        ai/lua/ca_recruit_rushers.lua:155: in local 'get_best_attack'
        ai/lua/ca_recruit_rushers.lua:285: in upvalue 'analyze_enemy_unit'
        ai/lua/ca_recruit_rushers.lua:986: in field 'execution'
        [string "local self, params, data, filter_own = ......"]:2: in main chunk
```

### Steps to reproduce the behavior

1. Open BfW client
2. Go to Multiplayer -> local game 
3. Pick any scenario (preferably a 1v1 for fastest selections)
4. Pick War of Legends as era.
5. Create Game
6. Pick one side to Experimental AI & set faction to "Deep Elves"
7. Leave other side(s) as human
8. Start game
9. Observe Error message on the AI side's turn

### Expected behavior

No error message and side can recruit units normally.

**Comment by ProditorMagnus** [comment]
AI assumes all units have configured defense on flat.

**Comment by knyghtmare** [comment]
> AI assumes all units have configured defense on flat.

I checked all units of the factions. They all have a flat defence on flat terrain. Just tell me if this is core bug or addon bug?

**Comment by ProditorMagnus** [comment]
Core bug in e08676d6a8a402f17214f92ce40a0ace5eb0e6a6. I think defense on flat is still requirement for ai to work, but that is not the failing situation in this faction.

**Comment by ProditorMagnus** [comment]
Compare
						wesnoth.message(defense - (tonumber(mod.sub) or 0))
						wesnoth.message(defense - tonumber(mod.sub) or 0)

**Comment by CelticMinstrel** [comment]
I looked through the code and couldn't find any way it could be nil, but I missed that minus has higher precedence than or. That has to be it.

**Comment by stevecotton** [comment]
I'm confused by the two things that have auto-linked. Is this already fixed in #8117 (which seems confirmed by a quick test), or is it still awaiting a fix cherry-picked from #7359?

**Comment by knyghtmare** [comment]
> I'm confused by the two things that have auto-linked. Is this already fixed in #8117 (which seems confirmed by a quick test), or is it still awaiting a fix cherry-picked from #7359?

If you read the comments in #8117, you would see that the fix did not work
and this is from latest master:
```ini
20240206 19:17:08 error scripting/lua: ai/lua/ca_recruit_rushers.lua:155: attempt to perform arithmetic on a nil value
stack traceback:
        ai/lua/ca_recruit_rushers.lua:155: in local 'get_best_attack'
        ai/lua/ca_recruit_rushers.lua:285: in upvalue 'analyze_enemy_unit'
        ai/lua/ca_recruit_rushers.lua:986: in field 'execution'
        [string "local self, params, data, filter_own = ......"]:2: in main chunk
20240206 19:17:08 error engine: failed to auto-store $unit at (-999,-999)


```

**Comment by ProditorMagnus** [comment]
The fix in #7359 does not work either, it puts () to wrong place.

**Comment by CelticMinstrel** [comment]
Whoops, I messed up there. Fixed.
